### ISSUE TITLE
Image replace not working as expected on Firefox

### ISSUE THREAD

**Issue by jackiejade** [Automattic/wp-calypso#44326]
#### Steps to reproduce

1. From within the page editor
2. Add a gallery block and select images
3. Insert the gallery
4. Decide to switch the images in the gallery - click "Media Library"
5. Remove two images and select two different in their place.
6. Click insert

#### What I expected

I expect the new images to replace the old images.

#### What happened instead

The old images remained. I had to try inserting the new images multiple times for it to take.

#### Browser / OS version
Firefox 78.0.2 / 10.14.6 Mojave

Also reported by a user in Firefox on Windows 10

#### Context / Source

This also happens when trying to replace an image in an image block.

Here is a gif of inserting a gallery from media library not working on first attemtp:

https://d.pr/i/LlJGA9

Here is a gif of replacing an image in the gallery not working:

https://d.pr/i/17zQAC

**Comment by jackiejade** [comment]
User reported this happening with a Featured image as well in 3170961-zen 
There is a video of the behavior in the ticket, but it has private user information from their site on it, so not sharing here.

**Comment by Copons** [comment]
Closing this as this is a duplicate of #44097.

tl;dr sometimes Firefox sends the wrong event when closing the media modal ("cancel" instead of "insert").

In case this might be of any help when supporting users, this issue doesn't seem to happen in Chrome.
